solr/core/src/java/org/apache/solr/core/CachingDirectoryFactory.java

```diff
@@ -391,25 +393,22 @@ public boolean exists(String path) throws IOException {
   public final Directory get(String path, DirContext dirContext, String rawLockType)
       throws IOException {
     String fullPath = normalize(path);
+    Directory directory;
+    CacheValue cacheValue;
     synchronized (this) {
       if (closed) {
         throw new AlreadyClosedException("Already closed");
       }
 
-      final CacheValue cacheValue = byPathCache.get(fullPath);
-      Directory directory = null;
-      if (cacheValue != null) {
-        directory = cacheValue.directory;
-      }
-
-      if (directory == null) {
+      cacheValue = byPathCache.get(fullPath);
+      if (cacheValue == null) {
         directory = create(fullPath, createLockFactory(rawLockType), dirContext);
```

Comment:
In a separate issue/PR, the `create` method should not contain the dirContext leaving a TODO now.  Or if you wish, make that change here; it's rather internal.

Reply:
I prefer to leave that for another PR (I have another proposal to make CachingDirectoryFactory support a DelegatingDirectoryFactory).
For the TODO, I'm unsure as I see some usage of the DirContext in HdfsDirectoryFactory which seems valid (creating either a BlockDirectory or a HdfsDirectory).

Reply:
Thanks! I add the todo.